pkg/controller/certificates/certificate_controller_test.go

```diff
@@ -64,7 +64,8 @@ func TestCertificateController(t *testing.T) {
 
 	stopCh := make(chan struct{})
 	defer close(stopCh)
-	go informerFactory.Start(stopCh)
+	informerFactory.Start(stopCh)
+	informerFactory.WaitForCacheSync(stopCh)
```

Comment:
the call to the eventhandler registered with the shared informer (which is what puts the item in the queue processed by `processNextWorkItem()`) is not synchronous. See https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/tools/cache/shared_informer.go#L547

instead, I'd loop until `controller.queue.Len() > 1`